src/main/java/org/apache/commons/lang3/CachedRandomBits.java

```diff
@@ -52,6 +52,16 @@ final class CachedRandomBits {
      */
     private int bitIndex;
 
+    /**
+     * The maximum size of the cache.
+     *
+     * <p>
+     * This is dictated by the {@code if (bitIndex >> 3 >= cache.length)} in the {@link #nextBits(int)} method.
+     * Essentially {@code 63_913_202 << 3 } will overflow so this line will never evaluate to true.
+     * At some point the {@code bitIndex += generatedBitsInIteration} will overflow and cause an exception.
+     */
+    private static final int MAX_CACHE_SIZE = 63_913_200;
```

Comment:
The value here should rather be `0x1FFF_FFFF` (i.e. `268_435_455`, but in hex it is easier to understand).

`63_913_201` is the highest `x` such that `(21x + 3) / 5 + 10 ≤ 268 435 455`.

Reply:
Fixed in https://github.com/apache/commons-lang/pull/1379/commits/659e14a74eea438ba0a58edb9c11886391ff3517